x/clp/keeper/grpc_query.go

```diff
@@ -105,21 +107,25 @@ func (k Querier) GetLiquidityProviderList(c context.Context, req *types.Liquidit
 	if req == nil {
 		return nil, status.Error(codes.InvalidArgument, "empty request")
 	}
-
 	ctx := sdk.UnwrapSDKContext(c)
 
 	searchingAsset := types.NewAsset(req.Symbol)
-	lpList := k.GetLiquidityProvidersForAsset(ctx, searchingAsset)
+	lpList, pageRes, err := k.GetLiquidityProvidersForAssetPaginated(ctx, searchingAsset, req.Pagination)
+	if err != nil {
+		return nil, err
+	}
+
+	liquidityProviders := make([]*types.LiquidityProvider, len(lpList))
 
-	lpl := make([]*types.LiquidityProvider, len(lpList))
 	for i, lp := range lpList {
 		lp := lp
```

Comment:
Do we want to make a copy here? All we want is to convert to pointers right?

Reply:
addressed it here #1673